editor/src/components/canvas/dom-walker.ts

```diff
@@ -53,12 +53,33 @@ import { PRODUCTION_ENV } from '../../common/env-vars'
 import { CanvasContainerID } from './canvas-types'
 import { emptySet } from '../../core/shared/set-utils'
 import { useForceUpdate } from '../editor/hook-utils'
+import { extractOriginalUidFromIndexedUid, getUIDsOnDomELement } from '../../core/shared/uid-utils'
+import { mapDropNulls } from '../../core/shared/array-utils'
+import { optionalMap } from '../../core/shared/optional-utils'
 
 const MutationObserverConfig = { attributes: true, childList: true, subtree: true }
 const ObserversAvailable = (window as any).MutationObserver != null && ResizeObserver != null
 
-function isValidPath(path: TemplatePath | null, validPaths: Array<string>): boolean {
-  return path != null && validPaths.indexOf(TP.toString(path)) > -1
+function findValidPath(uid: string | null, validPathsString: Array<string>): InstancePath | null {
+  const validPaths = validPathsString.map(TP.fromString) // Hi Rheese :)
+  if (uid == null) {
+    return null
+  }
+  return (
+    mapDropNulls((validPath) => {
+      if (TP.isScenePath(validPath)) {
+        return null // TODO do we need to care about a scene path here?
```

Comment:
I don't think so. I'm pretty sure the valid paths are always instance paths anyway